### PR TITLE
docs: Add timing for automatic agent updates to the cloud FAQ

### PR DESCRIPTION
**Pull request by lsgunn-teleport**
Fixes gravitational/customer-sensitive-requests#137

### CODE REVIEW

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx [hunk 1]
```diff
@@ -138,40 +138,46 @@ The ability to download recordings for offline viewing will be available in a fu
 
 ### Will Teleport be automatically upgraded with each release?
 
-Teleport Cloud follows the Teleport [Upcoming
+Teleport Team and Teleport Enterprise Cloud follow the Teleport [Upcoming
 Releases](../../upcoming-releases.mdx) timelines.
 
-- Major version upgrades for customer Auth and Proxy Servers occur after the
-  first minor release (for example (=teleport.major_version=).1.0) of Teleport.
+- Major version upgrades for cloud Teleport Auth Service and Teleport Proxy Service occur
```
Comment (ptgott) [inline docs/pages/choose-an-edition/teleport-cloud/faq.mdx:144]:
```suggestion
- Major version upgrades for the Teleport Auth Service and Teleport Proxy Service on Teleport Cloud occur 
```
Otherwise we're missing an article

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx [hunk 2]
```diff
@@ -138,40 +138,46 @@ The ability to download recordings for offline viewing will be available in a fu
...
   becoming available.
 - After the first minor release, subsequent minor and patch upgrades typically
   occur within a week.
 
 <Admonition type="note" title="Minimum agent version">
-    For major version upgrades (e.g. 12.x to 13.x), your Teleport tenant will NOT be automatically upgraded if any
-    Teleport agents connected to the cluster are not compatible (> 1 major version behind).  This is done to ensure
-    backwards compatibility. To get version information on your agents versions
-    see [here](#how-can-i-find-version-information-on-connected-agents).
+    For major version upgrades (for example, 12.x to 13.x), your Teleport tenant is NOT
```
Comment (ptgott) [inline docs/pages/choose-an-edition/teleport-cloud/faq.mdx:152]:
```suggestion
    For major version upgrades (for example, 12.x to 13.x), your Teleport account is NOT 
```
I think we were moving from "tenant" to "account"

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx [hunk 3]
```diff
@@ -138,40 +138,46 @@ The ability to download recordings for offline viewing will be available in a fu
...
 
 <Admonition type="note" title="Minimum agent version">
-    For major version upgrades (e.g. 12.x to 13.x), your Teleport tenant will NOT be automatically upgraded if any
-    Teleport agents connected to the cluster are not compatible (> 1 major version behind).  This is done to ensure
-    backwards compatibility. To get version information on your agents versions
-    see [here](#how-can-i-find-version-information-on-connected-agents).
+    For major version upgrades (for example, 12.x to 13.x), your Teleport tenant is NOT 
+    automatically upgraded if any Teleport agents connected to the cluster are not compatible 
+    (> 1 major version behind). This version check is done to ensure backwards compatibility. 
+    To get version information for your agents, see [here](#how-can-i-find-version-information-on-connected-agents).
```
Comment (ptgott) [inline docs/pages/choose-an-edition/teleport-cloud/faq.mdx:155]: Should we reword the "here" link text?

**Review by ptgott — APPROVED** [review]

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx (hunk 2 above)
Comment (lsgunn-teleport) [reply docs/pages/choose-an-edition/teleport-cloud/faq.mdx]: Calling it your "account" seems a little far removed from cluster/instance/tenant to me, but if that's the direction we're headed, that's fine.

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx [hunk 4]
```diff
@@ -138,40 +138,49 @@ The ability to download recordings for offline viewing will be available in a fu
 
 ### Will Teleport be automatically upgraded with each release?
 
-Teleport Cloud follows the Teleport [Upcoming
-Releases](../../upcoming-releases.mdx) timelines.
+Teleport Team and Teleport Enterprise Cloud are upgraded after each Teleport release
```
Comment (zmb3) [inline docs/pages/choose-an-edition/teleport-cloud/faq.mdx:141]: We don't necessarily upgrade to each release. I think we should generalize this statement a bit and just say something along the lines of:

- we keep your auth and proxy patched and up to date
- patches and minors can be applied as soon as the day after a release
- major upgrades are not applied immediately (1 month delay)

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx [hunk 5]
```diff
@@ -138,40 +138,49 @@ The ability to download recordings for offline viewing will be available in a fu
...
   becoming available.
 - After the first minor release, subsequent minor and patch upgrades typically
   occur within a week.
 
-<Admonition type="note" title="Minimum agent version">
-    For major version upgrades (e.g. 12.x to 13.x), your Teleport tenant will NOT be automatically upgraded if any
-    Teleport agents connected to the cluster are not compatible (> 1 major version behind).  This is done to ensure
-    backwards compatibility. To get version information on your agents versions
-    see [here](#how-can-i-find-version-information-on-connected-agents).
+For major version upgrades (for example, 12.x to 13.x), your Teleport account is NOT
```
Comment (zmb3) [inline docs/pages/choose-an-edition/teleport-cloud/faq.mdx:153]: This is correct today, but with automatic upgrades I believe we will no longer hold upgrades due to old agents. Would be good to confirm with @r0mant

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx [hunk 6]
```diff
@@ -138,40 +138,49 @@ The ability to download recordings for offline viewing will be available in a fu
...
+maintenance window start time. Click **Edit** to set a different start
+time for scheduled maintenance.
 
 Teleport recommends subscribing to the Teleport Enterprise Cloud updates at
 [status.teleport.sh](https://status.teleport.sh) to keep up to date on
 scheduled maintenance.
 
+### When are agents automatically updated?
+
+Typically, Teleport agents are automatically updated from one to a few days after
```
Comment (zmb3) [inline docs/pages/choose-an-edition/teleport-cloud/faq.mdx:181]: Let's make it clear that you must be enrolled in automatic agent upgrades (and probably link to those docs) for this to happen.

I like to think of auth/proxy upgrades as "push" upgrades - we (Teleport) push the upgrade during the configured maintenance window. Agent upgrades are "pull" upgrades - we aren't pushing anything out, the agents are just configured to periodically check for new versions and pull them down.

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx (hunk 5 above)
Comment (lsgunn-teleport) [reply docs/pages/choose-an-edition/teleport-cloud/faq.mdx]: @r0mant, can you elaborate on what we might need to say here?

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx [hunk 7]
```diff
@@ -136,42 +136,58 @@ The ability to download recordings for offline viewing will be available in a fu
...
+- Teleport agents are only upgraded automatically if you enroll in automatic agent updates.
 
-<Admonition type="note" title="Minimum agent version">
-    For major version upgrades (e.g. 12.x to 13.x), your Teleport tenant will NOT be automatically upgraded if any
-    Teleport agents connected to the cluster are not compatible (> 1 major version behind).  This is done to ensure
-    backwards compatibility. To get version information on your agents versions
-    see [here](#how-can-i-find-version-information-on-connected-agents).
+If you have Teleport agents connected to a Teleport Team or Teleport Enterprise cluster that are
+more than one major version behind, you might experience compatibility issues if your cluster is 
+automatically upgraded but agents aren't.
```
Comment (r0mant) [inline docs/pages/choose-an-edition/teleport-cloud/faq.mdx:154]: I would rephrase to:
```suggestion
more than one major version behind, you might experience compatibility issues unless your agents
are enrolled in automatic upgrades
```

**Review by r0mant — APPROVED** [review]

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx (hunk 5 above)
Comment (r0mant) [reply docs/pages/choose-an-edition/teleport-cloud/faq.mdx]: I can't see this paragraph fully anymore cause I assume it got removed but the current changeset looks good to me. Just left one comment.

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx [hunk 8]
```diff
@@ -136,42 +136,58 @@ The ability to download recordings for offline viewing will be available in a fu
...
+### When are agents automatically updated?
+
+Teleport Team and Teleport Enterprise must be set to receive automatic updates
+to use the Teleport Cloud version server for automatic agent updates. With automatic 
+agent upgrades, agents periodically check the version server for new releases and 
+update the software when new versions are found.
+
+If you enroll in automatic agent upgrades, Teleport agents are automatically 
+updated from one to a few days after your Teleport cluster is upgraded during 
+a scheduled maintenance period.
```
Comment (xinding33) [inline docs/pages/choose-an-edition/teleport-cloud/faq.mdx:189]: This should really just be "Teleport agents are automatically updated during your scheduled maintenance period."

**Review by xinding33 — APPROVED** [review]
No blocking changes but please take a look at the suggestions.

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx (hunk 8 above)
Comment (lsgunn-teleport) [reply docs/pages/choose-an-edition/teleport-cloud/faq.mdx:189]: @xinding33 just FYI, the issue implies that "during your scheduled maintenance period" isn't the case:

> Currently after a cloud upgrade the version for the agents can be updated 1 or multiple days after.
